[debug] [2018-07-27T18:24:46.713Z] ----------------------------------------------------------------------
[debug] [2018-07-27T18:24:46.717Z] Command:       /home/vagrant/.nvm/versions/node/v8.9.4/bin/node /home/vagrant/.nvm/versions/node/v8.9.4/bin/firebase login --no-localhost --debug
[debug] [2018-07-27T18:24:46.718Z] CLI Version:   4.0.0
[debug] [2018-07-27T18:24:46.719Z] Platform:      linux
[debug] [2018-07-27T18:24:46.719Z] Node Version:  v8.9.4
[debug] [2018-07-27T18:24:46.720Z] Time:          Fri Jul 27 2018 18:24:46 GMT+0000 (UTC)
[debug] [2018-07-27T18:24:46.720Z] ----------------------------------------------------------------------
[debug] 
[info] 
[info] Visit this URL on any device to log in:
[info] https://accounts.google.com/o/oauth2/auth?client_id=563584335869-fgrhgmd47bqnekij5i8b5pr03ho849e6.apps.googleusercontent.com&scope=email%20openid%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcloudplatformprojects.readonly%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Ffirebase%20https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fcloud-platform&response_type=code&state=926391524&redirect_uri=urn%3Aietf%3Awg%3Aoauth%3A2.0%3Aoob
[info] 
[debug] [2018-07-27T18:25:13.980Z] >>> HTTP REQUEST POST https://accounts.google.com/o/oauth2/token  
 { code: '4/AABEh_nqluHb1k9ePPulyG_UrqNdYWlvOZgAabPS7Mrvu-Nj9h4hoCw',
  client_id: '563584335869-fgrhgmd47bqnekij5i8b5pr03ho849e6.apps.googleusercontent.com',
  client_secret: 'j9iVZfS8kkCEFUPaAeJV0sAi',
  redirect_uri: 'urn:ietf:wg:oauth:2.0:oob',
  grant_type: 'authorization_code' } 
 Fri Jul 27 2018 18:25:13 GMT+0000 (UTC)
[debug] [2018-07-27T18:25:34.013Z] Token Fetch Error: Error
    at new FirebaseError (/home/vagrant/.nvm/versions/node/v8.9.4/lib/node_modules/firebase-tools/lib/error.js:11:16)
    at Request._callback (/home/vagrant/.nvm/versions/node/v8.9.4/lib/node_modules/firebase-tools/lib/api.js:35:11)
    at self.callback (/home/vagrant/.nvm/versions/node/v8.9.4/lib/node_modules/firebase-tools/node_modules/request/request.js:185:22)
    at emitOne (events.js:116:13)
    at Request.emit (events.js:211:7)
    at Request.onRequestError (/home/vagrant/.nvm/versions/node/v8.9.4/lib/node_modules/firebase-tools/node_modules/request/request.js:877:8)
    at emitOne (events.js:116:13)
    at ClientRequest.emit (events.js:211:7)
    at TLSSocket.socketErrorListener (_http_client.js:387:9)
    at emitOne (events.js:116:13)
[error] 
[error] Error: Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
